Source/carthage/Archive.swift

```diff
@@ -71,7 +71,7 @@ public struct ArchiveCommand: CommandType {
 					}
 				}
 				.collect()
-				.map { Array(Set($0)).sort() }
+				.map { Array(Set($0.filter{ $0.hasSuffix(".framework") })).sort() }
```

Comment:
Rather than use `filter` on the `Array`, you could use `filter` directly on the signal chain before the `collect()`.

```swift
.filter { $0.hasSuffix(".framework") }
.collect()
.map { Array(Set($0)).sort() }
```

But I think it'd be better to reuse the existing logic for whether a `BuildSettings` represents something that Carthage should build. There's a private `shouldBuildFrameworkType` function in `Xcode.swift` that we could make `internal`. 

Then this signal chain could use it and the `settings.frameworkType`—just like `shouldBuildScheme` does—to filter out settings that don't represent dynamic frameworks.

This will keep the code working in spite of unforeseen changes down the line and deduplicate logic.